### PR TITLE
mplhep

### PR DESCRIPTION
**Pull request by andrzejnovak**
Start moving plotting to interface through mplhep

### CODE REVIEW

**Comment by lgray** [comment]
@andrzejnovak All sorts of flake8 violations and you need to add mplhep to setup.py!

**Comment by lgray** [comment]
@andrzejnovak 

An alternative way to do this is:
```python
import matplotlib.pyplot as plt.
plt.switch_backend('agg')
```

which seems to be a bit more universal?

**Comment by lgray** [comment]
@andrzejnovak @nsmith- Now for proper code review. I'm guessing the interface discussion for mplhep has been happening in other channels?

**Comment by lgray** [comment]
@andrzejnovak @nsmith- What are the remaining items to attend to for this PR?

**Comment by andrzejnovak** [comment]
@lgray Looks like parsl issues in 3.6

File: tests/test_hist_plot.py [hunk 1]
```diff
@@ -169,7 +169,6 @@ def test_plotratio():
         'markersize': 10.,
         'color': 'k',
         'elinewidth': 1,
-        'emarker': '_'
```
Comment (nsmith-) [inline tests/test_hist_plot.py:162]: So this isn't supported anymore? It's a popular request to have caps on the ends of the errorbars.

Comment (nsmith-) [reply tests/test_hist_plot.py:162]: I had to add it manually in the old code (its not in mpl)

Comment (andrzejnovak) [reply tests/test_hist_plot.py:162]: Sure it is 
```
data_err_opts = {
    'linestyle':'none',
    'marker': '.',
    'markersize': 10.,
    'elinewidth': 1,
    'capsize':4
}
hist.plot.plot1d(kk, overlay='sample', error_opts=data_err_opts)
```
image

**Comment by lgray** [comment]
@andrzejnovak It's a transient bug in parsl, issue created here: https://github.com/Parsl/parsl/issues/1456

File: tests/test_hist_plot.py (hunk 1 above)
Comment (nsmith-) [reply tests/test_hist_plot.py:162]: heh ok, I could swear there were arrow caps by default on asymmetric errorbars when I was writing this, maybe it changed?

**Comment by andrzejnovak** [comment]
@lgray @nsmith- I guess this is now ready, tested a few more things and the plotting_demo example in unaffected by these changes

**Pull request merged**